src/components/ModelTableList/ModelTable/ModelTable.tsx

```diff
@@ -0,0 +1,208 @@
+import { MainTable } from "@canonical/react-components";
+import type { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+
+import ModelActions from "components/ModelActions";
+import Status from "components/Status";
+import TruncatedTooltip from "components/TruncatedTooltip";
+import { getActiveUsers, getControllerData } from "store/juju/selectors";
+import type { Controllers, ModelData } from "store/juju/types";
+import {
+  extractOwnerName,
+  getModelStatusGroupData,
+} from "store/juju/utils/models";
+import { useAppSelector } from "store/store";
+
+import CloudCell from "../CloudCell";
+import ModelNameCell from "../ModelNameCell";
+import ModelSummary from "../ModelSummary";
+import WarningMessage from "../StatusGroup/WarningMessage";
+import {
+  generateCloudAndRegion,
+  generateTableHeaders,
+  getControllerName,
+  getCredential,
+  getLastUpdated,
+  getRegion,
+} from "../shared";
+import { TestId } from "../types";
+
+/**
+  Returns the model info and statuses in the proper format for the table data.
+  @param  models The list of models
+  @return The formatted table rows.
+*/
+function generateModelTableList(
+  models: ModelData[],
+  activeUsers: Record<string, string>,
+  controllers: Controllers | null,
+  groupBy: "status" | "owner" | "cloud",
```

Comment:
It'd probably be a bit easier to maintain if these values were an enum so that we don't have to repeat this definition and also don't have to maintain magic strings.